### PR TITLE
Add Github Action for Autogenerate Java SDK

### LINKED ISSUE #21: Create GitHub Action to Dry-Run Autogenerate Java SDK Before PR Merge

### PR DESCRIPTION
**Pull request by madrigal-cqv**
**Description**

- Add Github Action for autogenerating Java SDK on pull request

**Motivation**

- Automated testing

**Testing Done**

- Tested on [push] event on a testing repo

**Backwards Compatibility Criteria (if any)**

N/A

### CODE REVIEW

File: java-generator-config.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+groupId: "com.branchoftriumph"
+invokerPackage: "com.branchoftriumph.llamacpp.openapi"
+modelPackage: "com.branchoftriumph.llamacpp.openapi"
+apiPackage: "com.branchoftriumph.llamacpp.openapi"
+library: okhttp-gson
+errorObjectType: "com.branchoftriumph.llamacpp.openapi.Error"
+annotationLibrary: "swagger2"
+groupId: "com.branchoftriumph"
```
Comment (ProfHugo) [inline java-generator-config.yml]: nit: duplicate groupId

File: .github/workflows/java-sdk-generator.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+      - name: Generate Client Library w/ the OpenAPITools Generator
+        uses: openapi-generators/openapitools-generator-action@v1.5.0
+        with:
+          generator: java
+          openapi-file: models/llamacpp.yml
+          config-file: java-generator-config.yml
+          command-args: -o llamacpp-openapi-javasdk --api-name-suffix 'Client'
```
Comment (ProfHugo) [inline .github/workflows/java-sdk-generator.yml]: can you check if ``--api-name-suffix 'Client'`` can be added to the config YAML file?

File: java-generator-config.yml (hunk 1 above)
Comment (madrigal-cqv) [reply java-generator-config.yml]: fixed

**Review by ProfHugo — APPROVED** [review]
LGTM

**Pull request merged**